modules/CmpdRegBulkLoader/src/server/routes/CmpdRegBulkLoaderRoutes.coffee

```diff
@@ -362,9 +364,49 @@ class HTTPResponseError extends Error
 		super("HTTP Error Response: #{response.status} #{response.statusText}", ...args)
 		this.response = response
 
+exports.getLotsByBulkLoadFileID = (req, resp) ->
+	req.connection.setTimeout 6000000
+	try
+		lotCorpNames = await exports.getLotsByBulkLoadFileIDInternal req.params.bulkFileID
+		resp.end JSON.stringify lotCorpNames
+	catch error
+		console.error error
+		resp.statusCode = 500
+		resp.end JSON.stringify error
+
+exports.getLotsByBulkLoadFileIDInternal = (bulkFileID) ->
+	config = require '../conf/compiled/conf.js'
+	baseurl = config.all.client.service.cmpdReg.persistence.fullpath+"bulkload/getLotsByBulkLoadFileID?bulkLoadFileID=" +bulkFileID
+	response = await fetch(baseurl)
+	checkStatus response
+	lotCorpNames = await response.json()
+	return lotCorpNames
+
 exports.getSDFFromBulkLoadFileId = (req, resp) ->
 	req.setTimeout 86400000
 	config = require '../conf/compiled/conf.js'
+	try
+		lotCorpNames = await exports.getLotsByBulkLoadFileIDInternal req.params.bulkFileID
+
+		# Use fetch to call the service and pipe the response to the client
+		baseurl = config.all.client.service.cmpdReg.persistence.fullpath+"export/lotCorpNames"
+		response = await fetch(baseurl,
+			method: 'POST'
+			body: JSON.stringify(lotCorpNames)
+			headers:
+				'Content-Type': 'application/json'
+		)
+		checkStatus response
+		resp.set({
+			"content-length": response.headers.get('content-length'),
+			"content-type": response.headers.get('content-type'),
+		})
+		response.body.pipe(resp);
+	catch err
+		console.log "Error calling service: " + err
+		resp.statusCode = 500
+		resp.json err
+
```

Comment:
This makes sense as a refactor to use your new `exports.getLotsByBulkLoadFileIDInternal`, but why is the old implementation still present below? Wouldn't we want to remove that?

Reply:
Thanks, missed that checkin, just pushed.